### PR TITLE
Support admin capacity migration  API  

### PR DESCRIPTION
**Pull request by toymsft**
Enable workspace  capacity migration using new public admin capacity migration API.

### Capacity
```
Set-PowerBIWorkspace [-Scope <PowerBIUserScope>] -Id <Guid> -CapacityId <Guid> [<CommonParameters>]
```

### CODE REVIEW

File: src/Modules/Workspaces/Commands.Workspaces/help/Set-PowerBIWorkspace.md [hunk 1]
```diff
@@ -51,6 +56,21 @@ If the current user is an administrator, this will update the specified workspac
 
 ## PARAMETERS
 
+### -CapacityId
+The capacity object id which a workspace will be assigined to. If CapacityId is set to "00000000-0000-0000-0000-000000000000", the workspace will be migrated to shared capacity.
```
Comment (nandemoii) [inline src/Modules/Workspaces/Commands.Workspaces/help/Set-PowerBIWorkspace.md]: assigined -> assigned

File: src/Modules/Workspaces/Commands.Workspaces.Test/WorkspacesTestUtilities.cs [hunk 2]
```diff
@@ -50,6 +55,16 @@ public static Workspace GetFirstWorkspaceInOrganization(System.Management.Automa
             return InvokeGetPowerBIWorkspace(ps, PowerBIUserScope.Organization, OperationType.GetFirstWorkspaceInOrganization);
         }
 
+        public static Guid GetFirstCapacityInOrganization(System.Management.Automation.PowerShell ps)
+        {
+            ps.AddCommand(GetPowerBICapacityCmdletInfo).AddParameter(nameof(GetPowerBICapacity.Scope), PowerBIUserScope.Organization); ;
```
Comment (nandemoii) [inline src/Modules/Workspaces/Commands.Workspaces.Test/WorkspacesTestUtilities.cs]: Double semicolon.

File: src/Common/Common.Api/Workspaces/WorkspacesClient.cs [hunk 3]
```diff
@@ -60,5 +60,34 @@ public object NewWorkspaceAsUser(string workspaceName)
...
             return (Workspace)this.Client.Groups.CreateGroup(new PowerBI.Api.V2.Models.GroupCreationRequest(name: workspaceName));
         }
+
+        public object MigrateWorkspaceCapacity(Guid workspaceId, Guid capacityId)
+        {
+            if (capacityId != Guid.Empty)
+            {
+                var request = new PowerBI.Api.V2.Models.AssignWorkspacesToCapacityRequest
+                {
+                    CapacityMigrationAssignments = new List<PowerBI.Api.V2.Models.CapacityMigrationAssignment>
```
Comment (nandemoii) [inline src/Common/Common.Api/Workspaces/WorkspacesClient.cs]: Looking closer at this class, I realize that we don't have a way for the cmdlets to take advantage of the bulk functionality available to us through these lists. Do you intend to add cmdlets later for this?

Comment (toymsft) [reply src/Common/Common.Api/Workspaces/WorkspacesClient.cs]: Currently in the spec we only support single workspace migration. Later we might add batch support.

File: src/Common/Common.Api/Workspaces/WorkspacesClient.cs [hunk 4]
```diff
@@ -60,5 +60,34 @@ public object NewWorkspaceAsUser(string workspaceName)
...
+                    {
+                        new PowerBI.Api.V2.Models.CapacityMigrationAssignment
+                        {
+                            WorkspacesToAssign = new List<string> { workspaceId.ToString() },
```
Comment (CodeCyclone) [inline src/Common/Common.Api/Workspaces/WorkspacesClient.cs]: Minor: I'm not sure what the contract is but I see List<string> used here and for unassign it uses string array. If we can make it consistent that would be good but not a requirement

File: src/Modules/Workspaces/Commands.Workspaces/SetPowerBIWorkspace.cs [hunk 5]
```diff
@@ -65,16 +71,28 @@ protected override void BeginProcessing()
 
         public override void ExecuteCmdlet()
         {
-            var workspaceId = this.ParameterSet.Equals(PropertiesParameterSetName) ? this.Id : this.Workspace.Id;
-            var updatedProperties = this.ParameterSet.Equals(PropertiesParameterSetName) ? new Workspace { Name = this.Name, Description = this.Description } : this.Workspace;
-
-            // The API will throw 400 saying that it "Cannot apply PATCH to navigation property users" if we don't null this property out
-            updatedProperties.Users = null;
+            if (this.ParameterSet.Equals(CapacityParameterSetName))
+            {
```
Comment (CodeCyclone) [inline src/Modules/Workspaces/Commands.Workspaces/SetPowerBIWorkspace.cs]: Minor: remove extra line

File: src/Modules/Workspaces/Commands.Workspaces/SetPowerBIWorkspace.cs [hunk 6]
```diff
@@ -65,16 +71,28 @@ protected override void BeginProcessing()
...
 
-            using (var client = this.CreateClient())
+                using (var client = this.CreateClient())
+                {
+                    client.Workspaces.MigrateWorkspaceCapacity(this.Id, this.CapacityId);
```
Comment (CodeCyclone) [inline src/Modules/Workspaces/Commands.Workspaces/SetPowerBIWorkspace.cs]: Ideally we support workspace object being passed in the pipeline too but it's not a requirement

Comment (CodeCyclone) [inline src/Modules/Workspaces/Commands.Workspaces/SetPowerBIWorkspace.cs]: Does this return a result?

File: src/Modules/Workspaces/Commands.Workspaces/help/Set-PowerBIWorkspace.md [hunk 7]
```diff
@@ -51,6 +56,21 @@ If the current user is an administrator, this will update the specified workspac
 
 ## PARAMETERS
 
+### -CapacityId
+The capacity object id which a workspace will be assigned to. If CapacityId is set to "00000000-0000-0000-0000-000000000000", the workspace will be migrated to shared capacity.
```
Comment (CodeCyclone) [inline src/Modules/Workspaces/Commands.Workspaces/help/Set-PowerBIWorkspace.md]: replace 00000000-0000-0000-0000-000000000000 with [guid]::Empty

File: src/Common/Common.Api/Workspaces/WorkspacesClient.cs (hunk 4 above)
Comment (toymsft) [reply src/Common/Common.Api/Workspaces/WorkspacesClient.cs]: Somehow I didn't get your comments and I merged it already. Let me open another PR to address this.

File: src/Modules/Workspaces/Commands.Workspaces/SetPowerBIWorkspace.cs (hunk 6 above)
Comment (toymsft) [reply src/Modules/Workspaces/Commands.Workspaces/SetPowerBIWorkspace.cs]: Somehow I didn't get your comments and I merged it already. Let me open another PR to address this.

Comment (toymsft) [reply src/Modules/Workspaces/Commands.Workspaces/SetPowerBIWorkspace.cs]: Currently there's no return value, I will update the SDK a little bit in order to have a return value.

File: src/Modules/Workspaces/Commands.Workspaces/help/Set-PowerBIWorkspace.md (hunk 7 above)
Comment (toymsft) [reply src/Modules/Workspaces/Commands.Workspaces/help/Set-PowerBIWorkspace.md]: Somehow I didn't get your comments and I merged it already. Let me open another PR to address this.

**Pull request merged**